src/notebooklm/_retry.py

```
@@ -0,0 +1,335 @@
+
+        Returns:
+            True if retries remain; False if all attempts exhausted.
+        """
+        if isinstance(error, RateLimitError) and error.retry_after is not None:
+            self._current_delay = float(error.retry_after)
+        else:
+            self._current_delay = min(
+                self._current_delay * self.backoff_factor, self.max_delay
+            )
+            if self.jitter:
+                self._current_delay *= 0.75 + random.random() * 0.5
```

Ruff (0.15.11)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)